addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -1,27 +1,415 @@
+        proxy_acks = []
+
+        # Save the attachments
+        attachments = {}
+        for id_transaction, invoice_data in invoices_data.items():
+            attachment = self._l10n_it_edi_store_download(invoice_data['filename'], invoice_data['file'], invoice_data['key'])
+            if attachment == self.env['ir.attachment']:
+                proxy_acks.append(id_transaction)
+            elif attachment:
+                attachments[id_transaction] = [invoice_data, attachment]
+
+        retrigger = False
+        Journal = self.env['account.journal'].with_context(default_move_type='in_invoice')
+        for id_transaction, (invoice_data, attachment) in attachments.items():
+
+            # The server has a maximum number of documents it can send at a time
+            # If that maximum is reached, then we search for more
+            # by re-triggering the download cron, avoiding the timeout.
+            current_num = invoices_data.get('current_num', 0)
+            max_num = invoice_data.get('max_num', 0)
+            retrigger = retrigger or current_num == max_num > 0
+
+            # Import the move from the attachment.
+            # `_create_document_from_attachment` will create an empty move
+            # then try and fill it with the content imported from the attachment.
+            # Should the import fail, thanks to try..except and savepoint,
+            # we will anyway end up with an empty `in_invoice` with the attachment posted on it.
+            # The move will be committed anyway
+            Journal._create_document_from_attachment(attachment.ids)
+            self.env.cr.commit()
+            proxy_acks.append(id_transaction)
+
+        return {"retrigger": retrigger, "proxy_acks": proxy_acks}
+
+    def _l10n_it_edi_get_moves_map(self, move=None):
+        if move:
+            return {move.l10n_it_edi_transaction: move}
+        else:
+            moves_to_update = self.env['account.move']._l10n_it_edi_search_moves_waiting_update(self.company_id)
+            return {m.l10n_it_edi_transaction: m for m in moves_to_update}
+
+    def _l10n_it_edi_demo_mode_update(self, id_transaction_to_move_map):
+        self.ensure_one()
+        for id_transaction, move_to_update in id_transaction_to_move_map.items():
+            filename = move_to_update.l10n_it_edi_attachment_id and move_to_update.l10n_it_edi_attachment_id.name or '???'
+            self._l10n_it_edi_process_update(
+                move_to_update,
+                id_transaction,
+                transformed_update_data={
+                    'l10n_it_edi_state': 'forwarded',
+                    'l10n_it_edi_transaction': f'demo_{uuid.uuid4()}',
+                    'send_ack_to_edi_proxy': False,
+                    'date': fields.Date.today(),
+                    'filename': filename},
+                message=_("The e-invoice file %s has been sent in Demo EDI mode.", filename))
+
+    def _l10n_it_edi_update(self, move=None, interactive=False):
```

Comment:
Every method that directly calls the proxy is in the account_edi_proxy_user file.
I will move these functions to the move itself, I do agree that it might be a better place.
![image](https://github.com/odoo/odoo/assets/1665365/0eeeb30e-fcd5-487c-982b-71ff50bddd6a)

Reply:
This actually improves the code a lot.
There was a lot of stupid passing of id_transactions, while all I need is on the move itself.